src/scripts/themed_posts.js

```diff
@@ -0,0 +1,65 @@
+import { buildStyle, filterPostElements, blogViewSelector } from '../util/interface.js';
+import { getPreferences } from '../util/preferences.js';
+import { onNewPosts } from '../util/mutations.js';
+import { timelineObject } from '../util/react_props.js';
+
+const styleElement = buildStyle();
+const blogs = new Set();
+let blacklist;
+
+const defaultBackgroundColor = '#FAFAFA';
+
+const hexToRGB = (hex) => {
+  const { red, green, blue } = hex.match(/^#(?<red>[A-Fa-f0-9]{1,2})(?<green>[A-Fa-f0-9]{1,2})(?<blue>[A-Fa-f0-9]{1,2})$/).groups;
+  return [red, green, blue]
+    .map(color => color.padEnd(2, color))
+    .map(color => parseInt(color, 16))
+    .join(', ');
+};
+
+const processPosts = async function (postElements) {
+  filterPostElements(postElements, { includeFiltered: true }).forEach(async postElement => {
+    if (postElement.matches(blogViewSelector)) return;
+
+    const { blog: { name, theme: { backgroundColor, titleColor } } } =
+      await timelineObject(postElement);
+
+    if (blacklist.includes(name)) return;
+
+    let backgroundColorRGB = hexToRGB(backgroundColor);
+    const titleColorRGB = hexToRGB(titleColor);
+
+    if (backgroundColor === defaultBackgroundColor) {
+      backgroundColorRGB = '255, 255, 255';
+    }
+
+    if (!blogs.has(name)) {
+      blogs.add(name);
+      styleElement.textContent += `
+        [data-xkit-themed="${name}"] {
+          --white: ${backgroundColorRGB};
+          --black: ${titleColorRGB};
+        }
+      `;
+    }
```

Comment:
it looks like this is calculating RGB values and then discarding them if the blog is already registered

```suggestion
    const { blog: { name, theme } } = await timelineObject(postElement);

    if (blacklist.includes(name)) return;

    if (!blogs.has(name)) {
      blogs.add(name);

      const { backgroundColor, titleColor } = theme;

      const backgroundColorRGB = hexToRGB(backgroundColor);
      const titleColorRGB = hexToRGB(titleColor);

      styleElement.textContent += `
        [data-xkit-themed="${name}"] {
          --white: ${backgroundColorRGB};
          --black: ${titleColorRGB};
        }
      `;
    }

    postElement.dataset.xkitThemed = name;
```